community/values/src/main/java/org/neo4j/values/storable/UTF8StringValue.java

```diff
@@ -0,0 +1,403 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.values.storable;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+
+/*
+ * Just as a normal StringValue but is backed by a byte array and does string
+ * serialization lazily when necessary.
+ *
+ */
+public final class UTF8StringValue extends StringValue
+{
+    //0111 1111, used for removing HIGH BIT from byte
+    private static final int HIGH_BIT_MASK = 127;
```

Comment:
How about using binary literals, and a javadoc comment:

    /** Used for removing the high order bit from byte. */
    private static final int HIGH_BIT_MASK = 0b0111_1111;